netbox/core/signals.py

```diff
@@ -182,6 +181,20 @@ def clear_events_queue(sender, **kwargs):
 # DataSource handlers
 #
 
+@receiver(post_save, sender=DataSource)
+def enqueue_sync_job(instance, created, **kwargs):
+    """
+    When a DataSource is saved, check its sync_interval and enqueue a sync job if appropriate.
+    """
+    from .jobs import SyncDataSourceJob
+
+    if instance.sync_interval:
+        SyncDataSourceJob.enqueue_once(instance, interval=instance.sync_interval)
+    elif not created:
+        # Delete any previously scheduled recurring jobs for this DataSource
+        SyncDataSourceJob.get_jobs(instance).filter(interval__isnull=False).delete()
```

Comment:
Ran in to what seem like edge cases while testing this:

1. when instance is marked disabled and saved, already queued recurring job is still run
2. when sync_interval is changed to None and saved, existing job records are deleted, even if the completed in one way or another. (Not sure if this one is intended or not.)
3. when sync_interval is changed to None (from Minutely in my case) saved, tasks still keeps executing in the worker every minute

Reply:
Good catches! Appreciate the thorough review.

Reply:
Oooh, found one last edge case I missed before. Although this may just fall under expected behavior, I think it could be confusing for the user:
1. Create a DataSource with a sync interval and is enabled
2. Manually delete the scheduled job

At this point any/all scheduled jobs are gone but the DataSource instance still shows it is enabled and scheduled (minutely in my test case). I could see that leading users to wonder why it indicates it should be running but is not.

Having said that, editing and saving the DataSource instances without making any changes does successfully re-schedule the recurring job.

So ... yeah, not sure this edge case is worth addressing. Might be better to just document that this is possible.